Dockerfile

```diff
@@ -0,0 +1,90 @@
+# syntax=docker/dockerfile:1
+
+ARG BASE_IMAGE=ubuntu:22.04
+FROM ${BASE_IMAGE} AS builder
+
+RUN apt update && apt install -y --no-install-recommends \
+    python3 \
+    git \
+    ca-certificates \
+    sudo \
+    make \
+    cmake \
+    g++ \
+    libevent-dev \
+    zlib1g-dev
+
+# ---
+
+FROM builder AS deps
+
+ARG PHOSG_TARGET=master
+ARG RESOURCE_DASM_TARGET=master
+ARG BUILD_RESOURCE_DASM=true
+
+RUN git clone --depth 1 -b ${PHOSG_TARGET} https://github.com/fuzziqersoftware/phosg.git && \
+    cd phosg && \
+    sed -i 's/VERSION 3\.[0-9]\+/VERSION 3.10/' CMakeLists.txt && \
```

Comment:
This is unfortunate. Is it necessary because 22.04 doesn't have a sufficiently new version of CMake? Could we use 24.04 for the base image instead?

Reply:
Ubuntu 22.04 uses CMake 3.22 , while the projects request 3.27 .
However, Ubuntu 24.04 uses `gcc-13` instead of `gcc-11` and fails to build due to `-Werror` params.
I think I can try to fix this in another PR...

Reply:
```
/usr/src/newserv/src/Episode3/RulerServer.cc: In member function 'int32_t Episode3::RulerServer::error_code_for_client_setting_card(uint8_t, uint16_t, const Episode3::Location*, uint8_t) const':
/usr/src/newserv/src/Episode3/RulerServer.cc:1870:7: error: 'x_offset' may be used uninitialized [-Werror=maybe-uninitialized]
 1870 |       if (x_offset == 0) {
      |       ^~
/usr/src/newserv/src/Episode3/RulerServer.cc:1868:15: note: 'x_offset' was declared here
 1868 |       int32_t x_offset, y_offset;
      |               ^~~~~~~~
```

Reply:
fixed with 75672a7 , please check?

Reply:
Thanks for taking care of this.